### ISSUE TITLE
Feature/Documentation: invalidateFrameloop support

### ISSUE THREAD

**Issue by FunMiles** [protectwise/troika#77]
My Canvas in three-fiber uses `invalidateFrameloop` so that the rendering is not continuous.
I am using a list of `Text` elements that changes when the user types some input in a separate `TextInput`.
When some text is input, under some conditions, a new `Text` is added via a state change, triggering a render. If the added `Text` uses letters already present in any of the other `Text`s, then the rendering is immediate. Otherwise, they will only appear later when manually changing the view and thus triggering a render.

From my understanding `Text` depends on async work to get the font and create the SDF. Unfortunately there is no call-back support to notify when it is ready in order to trigger a redraw of the Canvas. 
The documentation talks about preloading fonts but does not detail if this can be used for on a need basis drawing.
Is it doable, or should a new call-back property be added?

Edit: The idea that if letters have already been seen, then the Text appears is not always consistently the case... Maybe some other async issue.

**Comment by lojjic** [comment]
Are you using the abstraction from `drei`? AFAICT that already calls invalidate in their `sync()` callback, so I'd think that would be working.

Anyway, there are two ways to respond to the asynchronous result:
1. Pass a callback function to the `sync()` method call when you invoke it, or
2. Add an event listener on the Text mesh for the `synccomplete` event.

This would be good to add to the docs.

**Comment by FunMiles** [comment]
I am using `drei` for `OrbitControl` but not for `Text`. I forgot what happened but it didn't work correctly when I first tried.
However maybe my initial problem was elsewhere and I can try again.
For info, I use:
```javascript
import { Text, preloadFont } from "troika-three-text";
extend({Text});
```

**Comment by FunMiles** [comment]
@lojjic Can you give an example of the event listener for `synccomplete` with the simple `import/extend` I use?

**Comment by lojjic** [comment]
In pure ThreeJS code you'd do:

```
const myText = new Text()
myText.addEventListener('synccomplete', () => {
  //do stuff after sync...
})
```

I'm not sure if R3F exposes a declarative way to add event listeners like that. You may need to wrap it in a React component. Though, that's what drei's component is supposed to be anyway.

Looks like they have an issue for the dispose error: https://github.com/react-spring/drei/issues/121

**Comment by FunMiles** [comment]
That issue is just 5 days old. Interesting. I did not want to open an issue when I encountered the problem because I was too new to `drei`.
Maybe I'll try wrapping your code. I still have to resole how to propagate the event back to the rendering. I'll have to look at that when I have more time over the weekend.

**Comment by lojjic** [comment]
I've added this to the docs, and I think that gets you what you'd need to properly invalidate your frame loop. So I'm gonna close this issue. Reopen if you disagree. :)

**Comment by FunMiles** [comment]
@lojjic I saw the fix you made for `drei`. I tried to copy the changed file to my `react-create-app` directory but it didn't work. I must say that I'm a bit of a novice on the react side of things... Are you pushing an update I can install with `npm update`?

**Comment by lojjic** [comment]
That fix is in 0.32.0. But drei will have to bump their dependency to use it.
